src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -81,6 +81,39 @@ public static BooleanExpression isFilePresentForSelectedEntry(StateManager state
         return BooleanExpression.booleanExpression(fileIsPresent);
     }
 
+    /**
+     * Check if at least one of the selected entries has linked files
+     * <br>
+     * Used in {@link org.jabref.gui.maintable.OpenExternalFileAction} when multiple entries selected
+     * @param stateManager manager for the state of the GUI
+     * @param preferencesService services of preferences
+     * @return a boolean binding
+     */
+    public static BooleanExpression hasPresentFileForSelectedEntries(StateManager stateManager, PreferencesService preferencesService) {
+        return Bindings.createBooleanBinding(() -> {
+            ObservableList<BibEntry> selectedEntries = stateManager.getSelectedEntries();
+            List<LinkedFile> files;
+            for (BibEntry entry:selectedEntries) {
+                files = entry.getFiles();
+
+                if ((entry.getFiles().size() > 0) && stateManager.getActiveDatabase().isPresent()) {
+                    if (files.get(0).isOnlineLink()) {
```

Comment:
I would exclude online file links. Otherwise you will suddenly have 300 Tabs open if you select 300 entries and they have at least one link

Reply:
What's the difference to selecting 300 entries that have pdf attached?

Reply:
Should we set a limit?
For example, if a user is trying to open 100 files, pop out a dialog to ask the user whether to continue or cancel.
But how to determine the limit.

Reply:
Yep, that's a good idea. "You are about to open more than 10 files". Continue?
I would simply create a constant with 10. I think that's appropriate.
  - Continue /  Cancel

Reply:
OK.